### PR TITLE
core: spmc: fix FIP SP loading

### PR DESCRIPTION
**Pull request by balint-dobszay-arm**
The memory management in process_sp_pkg() function contains errors. It tries to add new mappings for the SP packages that reside in the TA_RAM PA range, but this range is already mapped so this is unnecessary and wrong. Fix the code by simply using phys_to_virt() instead.

### CODE REVIEW

File: core/arch/arm/kernel/secure_partition.c [hunk 1]
```diff
@@ -1645,11 +1612,6 @@ static TEE_Result sp_init_all(void)
 				panic();
 		}
 	}
-	/*
-	 * At this point all FIP SPs are loaded by ldelf so the original images
-	 * (loaded by BL2 earlier) can be unmapped
-	 */
-	fip_sp_unmap_all();
```
Comment (jforissier) [inline core/arch/arm/kernel/secure_partition.c:1652]: IIUC moving this call up (before the the `sp_first_run()` calls) is unrelated to this fix, in which case it would be better to create a separate commit.

**Review by jforissier — COMMENTED** [review]
Hi @balint-dobszay-arm,

One small comment, in any case LGTM:

`Acked-by: Jerome Forissier <jerome.forissier@linaro.org>`

File: core/arch/arm/kernel/secure_partition.c (hunk 1 above)
Comment (balint-dobszay-arm) [reply core/arch/arm/kernel/secure_partition.c:1652]: Okay, I'll move this to a separate commit.

File: core/arch/arm/kernel/secure_partition.c [hunk 2]
```diff
@@ -1418,111 +1418,76 @@ static const struct ts_ops sp_ops = {
...
-	tee_mm_entry_t *mm = NULL;
 	struct fip_sp *sp = NULL;
 	uint64_t sp_fdt_end = 0;
 	size_t sp_pkg_size = 0;
 	vaddr_t sp_pkg_va = 0;
-	size_t num_pages = 0;
 
-	/* Map only the first page of the SP package to parse the header */
+	/* Check the first page which contains the SP package header */
 	if (!tee_pbuf_is_sec(sp_pkg_pa, SMALL_PAGE_SIZE))
```
Comment (jenswi-linaro) [inline core/arch/arm/kernel/secure_partition.c:1429]: This isn't needed, `phys_to_virt()` below will fail if the PA isn't in the expected range.

File: core/arch/arm/kernel/secure_partition.c [hunk 3]
```diff
@@ -1418,111 +1418,76 @@ static const struct ts_ops sp_ops = {
...
-		goto err_unmap;
+		return TEE_ERROR_BAD_FORMAT;
 	}
 
-	core_mmu_unmap_pages(sp_pkg_va, 1);
-	tee_mm_free(mm);
-
-	/* Map the whole package */
+	/* Check the whole SP package now that the size is known */
 	if (!tee_pbuf_is_sec(sp_pkg_pa, sp_pkg_size))
```
Comment (jenswi-linaro) [inline core/arch/arm/kernel/secure_partition.c:1464]: This is also taken care of implicitly by `phys_to_virt()` below.

**Review by jenswi-linaro — COMMENTED** [review]
With the comments addressed:
`Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>`

File: core/arch/arm/kernel/secure_partition.c (hunk 2 above)
Comment (balint-dobszay-arm) [reply core/arch/arm/kernel/secure_partition.c:1429]: Thanks, fixed.

File: core/arch/arm/kernel/secure_partition.c (hunk 3 above)
Comment (balint-dobszay-arm) [reply core/arch/arm/kernel/secure_partition.c]: Fixed

**Comment by balint-dobszay-arm** [comment]
Thanks for the review, all comments addressed and tags applied.
